src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHost.cs

```diff
@@ -148,6 +149,8 @@ internal async Task RunAsyncCore(CancellationToken cancellationToken, WebAssembl
             var jsComponentInterop = new JSComponentInterop(_rootComponents.JSComponents);
             _renderer = new WebAssemblyRenderer(Services, loggerFactory, jsComponentInterop);
 
+            WebAssemblyNavigationManager.Instance.CreateLogger(loggerFactory);
```

Comment:
I'm not a fan of having to do it this way, but it seemed like one of the only options given the `WebAssemblyNavigationManager` instance is a static singleton unassociated with DI. I'm definitely open to alternative mechanisms for logging exceptions thrown in "location changing" events.

Reply:
I don't remember why it is done this way, maybe we could change it to be created from DI and that way it only needs to get injected in the constructor.